shadowsocks/dialer_test.go

```diff
@@ -0,0 +1,212 @@
+package shadowsocks
+
+import (
+	"io"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-ss-server/metrics"
+	onet "github.com/Jigsaw-Code/outline-ss-server/net"
+	"github.com/shadowsocks/go-shadowsocks2/shadowaead"
+	"github.com/shadowsocks/go-shadowsocks2/socks"
+)
+
+const (
+	testCipher   = "chacha20-ietf-poly1305"
+	testPassword = "testPassword"
+	testPayload  = "!!!test~payload!!!"
+)
+
+func TestShadowsocksDialer_DialTCP(t *testing.T) {
+	// Let the OS pick an available port
+	listenTCPAddr := &net.TCPAddr{IP: net.ParseIP("127.0.0.1"), Port: 0}
+	proxyListener, err := net.ListenTCP("tcp", listenTCPAddr)
```

Comment:
Can you move this to startShadowsocksTCPProxy?

Reply:
Done.